src/misc/messages.cpp

```diff
@@ -62,56 +83,78 @@ class Message {
 		if (rendered_msg.empty())
 			rendered_msg = convert_ansi_markup(markup_msg.c_str());
 		assert(rendered_msg.length());
-		return rendered_msg.c_str();
+
+		if (!is_utf8_msg)
```

Comment:
Unfortunately, it can't do this with 100% confidence. The proper solution would be to:
- either convert all the strings passed to `MSG_Add` to UTF-8
- or consider them CP437 characters and convert them to UTF-8
I intend to do one of these in a future PR.